core/federated/RTI/rti_remote.c

```diff
@@ -1323,6 +1318,7 @@ static int receive_connection_information(int* socket_id, uint16_t fed_id) {
 
     // Allocate memory for the upstream and downstream pointers
     if (fed->enclave.num_upstream > 0) {
+      // FIXME: This looks wrong. We are casting an array of uint16_ts to an int*?
```

Comment:
I fixed it. What makes me a little weary is that this should have lead to out-of-bounds writes. We essentially allocate only half of the memory we need (we allocate for uint16 but interpret it as ints). So this would lead to writing outside the memory returned by malloc. It is hard for me to believe that this would not cause weird issues...

Reply:
I was able to confirm the memory error with valgrind. So thanks for pointing it out, this was a segmentation fault/memory corruption error waiting to happen